addons/mrp/models/product.py

```diff
@@ -147,8 +147,12 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
         # pre-compute bom lines and identify missing kit components to prefetch
         bom_sub_lines_per_kit = {}
         prefetch_component_ids = set()
+        exploded = {}
         for product in bom_kits:
-            __, bom_sub_lines = bom_kits[product].explode(product, 1)
+            bom = bom_kits[product]
+            if bom not in exploded:
+                exploded[bom] = bom.explode(product, 1)
```

Comment:
It looks wrong.

It's the same BoM, but exploded by different products. You can't cache only the BoM.

Reply:
Yep, I was re-reviewing it and something seemed off